test/testmathlib.cpp

```diff
@@ -312,6 +313,11 @@ class TestMathLib : public TestFixture {
         ASSERT_EQUALS_DOUBLE(0.0   , MathLib::toDoubleNumber("-0.0"));
         ASSERT_EQUALS_DOUBLE(0.0   , MathLib::toDoubleNumber("+0.0"));
 
+        // Ticket #6795
+        std::setlocale(LC_NUMERIC, "de_DE");
```

Comment:
Be careful about performance. I've seen local-switching as a time-consuming task.